### PR TITLE
Add twoWayKey checks and multiple many-to-many restrictions

### PR DESCRIPTION
**Pull request by fogelito**
Fix for relations attributes when twoWayKey variable is not passed in API,
We do not want the worker to throw the Error
Until now, it was checked only for twoWay = true

Added a validation to prevent the creation of 2 manyToMany relations between the same collections

https://github.com/appwrite/appwrite/issues/6281

### CODE REVIEW

File: app/controllers/api/databases.php [hunk 1]
```diff
@@ -1640,8 +1640,55 @@ function updateAttribute(
         Event $queueForEvents
```
Comment (fogelito) [inline app/controllers/api/databases.php:1640]: @abnegate Since these are only relation validations, maybe should be cleaner to leave it here, for example, the loop through the attributes we do not have in the createAttribute function 
What do you think

File: app/controllers/api/databases.php [hunk 2]
```diff
@@ -1640,8 +1640,55 @@ function updateAttribute(
...
     ) {
         $key ??= $relatedCollectionId;
+        $twoWayKeyNull = is_null($twoWayKey);
```
Comment (stnguyen90) [inline app/controllers/api/databases.php:1643]: This is never used.

File: app/controllers/api/databases.php [hunk 3]
```diff
@@ -1640,8 +1640,55 @@ function updateAttribute(
...
+                throw new Exception(Exception::ATTRIBUTE_ALREADY_EXISTS);
+            }
+
+            if (
+                $attribute->getAttribute('type') === Database::VAR_RELATIONSHIP &&
+                \strtolower($attribute->getAttribute('options')['twoWayKey']) === \strtolower($twoWayKey) &&
+                $attribute->getAttribute('options')['relatedCollection'] === $relatedCollection->getId()
+            ) {
+                // Currently, we always throw an Exception even when, We do not want to change $twoWayKsy on $twoWayKeyNull
+                throw new Exception(Exception::ATTRIBUTE_ALREADY_EXISTS);
```
Comment (stnguyen90) [inline app/controllers/api/databases.php:1679]: Can we provide a clearer error message?

File: app/controllers/api/databases.php [hunk 4]
```diff
@@ -1640,8 +1640,55 @@ function updateAttribute(
...
+            }
+
+            if (
+                $type === Database::RELATION_MANY_TO_MANY &&
+                $attribute->getAttribute('type') === Database::VAR_RELATIONSHIP &&
+                $attribute->getAttribute('options')['relationType'] === Database::RELATION_MANY_TO_MANY &&
+                $attribute->getAttribute('options')['relatedCollection'] === $relatedCollection->getId()
+            ) {
+                throw new Exception(Exception::ATTRIBUTE_ALREADY_EXISTS, 'Creating more than one "manyToMany" relationship on the same collection is currently not permitted.');
+            }
```
Comment (stnguyen90) [inline app/controllers/api/databases.php:1689]: Is this supposed to address https://github.com/appwrite/appwrite/issues/6887? If so, i think it should be a separate PR linked to that https://github.com/appwrite/appwrite/issues/6887. We should probably improve these junction tables to:

1. properly allow multiple
2. make sure they work with multiple databases

File: app/controllers/api/databases.php (hunk 2 above)
Comment (fogelito) [reply app/controllers/api/databases.php:1617]: Yep removed..
In the beginning, we wanted to auto-generate a key we realized that was not a good idea.

File: app/controllers/api/databases.php (hunk 3 above)
Comment (fogelito) [reply app/controllers/api/databases.php:1653]: Done!

File: app/controllers/api/databases.php (hunk 4 above)
Comment (fogelito) [reply app/controllers/api/databases.php:1663]: Currently we do not have this option in Utopia , Definitely something to improve in the future.
The issue you mention here sounds like the reason we are restricting this for now.

File: app/controllers/api/databases.php [hunk 5]
```diff
@@ -1640,8 +1640,55 @@ function updateAttribute(
...
+        $relatedCollection = $dbForProject->getCollection('database_' . $databaseDocument->getInternalId() . '_collection_' . $relatedCollectionDocument->getInternalId());
+
+        if ($relatedCollection->isEmpty()) {
+            throw new Exception(Exception::COLLECTION_NOT_FOUND);
+        }
+
+        $attributes = $collection->getAttribute('attributes', []);
+        /** @var Document[] $attributes */
+        foreach ($attributes as $attribute) {
+            if (\strtolower($attribute->getId()) === \strtolower($key)) {
```
[the remark this answers is not in the document]
Comment (fogelito) [reply app/controllers/api/databases.php:1646]: Done!

File: app/controllers/api/databases.php [hunk 6]
```diff
@@ -1640,8 +1640,55 @@ function updateAttribute(
         Event $queueForEvents
     ) {
         $key ??= $relatedCollectionId;
+        $twoWayKeyNull = is_null($twoWayKey);
         $twoWayKey ??= $collectionId;
 
+        $databaseDocument = Authorization::skip(fn() => $dbForProject->getDocument('databases', $databaseId));
```
[the remark this answers is not in the document]
Comment (fogelito) [reply app/controllers/api/databases.php]: Done!
